Yoast/wordpress-seo
# Refactor the indexing implementation

## Context


*

## Summary


This PR can be summarized in the following changelog entry:

* Refactors the indexing implementation.

## Relevant technical choices:

* Renames the `indexing-integration` to `indexing-tools-integration`, a class that has as its responsibility to present the button on the tools page (if there are enough unindexed items) and pass it the information it needs.
* Renames the `indexing-indexables-integration` to `background-indexing-integration`, and makes it have a single responsibility (the background indexing). The endpoints have been moved to the `indexing-tool-integration`.
* Splits the `complete` action into two separate actions: the `indexable-indexing-complete-action`, which is executed after the indexing of the indexables has completed, and the `indexing-complete-action`, which is executed after all the indexing processes have completed.
* Removes the `indexing-links-integration` (its code has been moved to `indexing-tools-integration`).
* Similarly, removes the `links-indexing-route` (its code has been moved to `indexing-route`. There was no reason to have this in separate classes).

## Test instructions

This PR can be tested by following these steps:

The 'Start SEO data optimization' button and its notification should work properly. A non-exhaustive list of the things that should work:
* The progress bar on the Yoast Tools page should fill itself while the indexing takes place.
* The `wp_yoast_indexable`, `wp_yoast_indexable_hierarchy` and `wp_seo_links` tables should be filled with data when the indexing process is complete.
* A notification should be shown on the Yoast dashboard when there are more than 25 items that should be indexed (including on fresh installs of Yoast SEO).

## UI changes

## Documentation

## Quality assurance


Fixes #
